test/format/backup.c

```diff
@@ -286,25 +286,32 @@ copy_blocks(WT_SESSION *session, WT_CURSOR *bkup_c, const char *name)
 
                 first_pass = false;
             }
-            if (tmp_sz < size) {
-                tmp = drealloc(tmp, size);
-                tmp_sz = size;
+            this_size = WT_MIN(size, BACKUP_MAX_COPY);
+            if (tmp_sz < this_size) {
+                tmp = drealloc(tmp, this_size);
+                tmp_sz = this_size;
+            }
+            total = 0;
+            while (total < size) {
+                /*
+                 * Don't use the system checker for lseek. The system check macro uses an int which
+                 * is often 4 bytes and checks for any negative value. The offset returned from
+                 * lseek is 8 bytes and we can have a false positive error check.
+                 */
+                if (lseek(rfd, (wt_off_t)offset, SEEK_SET) == -1)
+                    testutil_die(errno, "backup-read: lseek");
+                error_sys_check(rdsize = read(rfd, tmp, this_size));
+                if (lseek(wfd1, (wt_off_t)offset, SEEK_SET) == -1)
+                    testutil_die(errno, "backup-write1: lseek");
+                if (lseek(wfd2, (wt_off_t)offset, SEEK_SET) == -1)
+                    testutil_die(errno, "backup-write2: lseek");
+                /* Use the read size since we may have read less than the granularity. */
+                error_sys_check(write(wfd1, tmp, (size_t)rdsize));
+                error_sys_check(write(wfd2, tmp, (size_t)rdsize));
+                total += this_size;
```

Comment:
`total += rdsize` so if we get a short read we don't skip anything.  Likewise for offset.

Reply:
Good catch. I had thought of it as I was making the changes and then forgot to go back and fix that. Fixed now.